### PR TITLE
#342 Update Three.js to the latest version

### PR DESCRIPTION
**Pull request by r1z3rISGOD**

### CODE REVIEW

File: packages/lib/src/gfx/shaders/UberMaterial.js [hunk 1]
```diff
@@ -197,187 +121,257 @@ UberMaterial.prototype.uberOptions = {
...
+    const { precision } = capabilities;
+    const str = `precision ${precision} float;\n`
+      + `precision ${precision} int;\n\n`;
+    return str;
+  }
 
-  this.uberOptions.copy(source.uberOptions);
+  // create copy of this material
+  copy(source) {
+    THREE.RawShaderMaterial.prototype.copy.call(this, source);
```
Comment (AnaMoskvina) [inline packages/lib/src/gfx/shaders/UberMaterial.js:195]: can we use super.copy(source) here?

File: packages/lib/src/gfx/shaders/UberMaterial.js [hunk 2]
```diff
@@ -197,187 +121,257 @@ UberMaterial.prototype.uberOptions = {
...
+    // mark as thick lines
+    this.thickLine = false;
+    // makes fog begin transparency (required for transparent background)
+    this.fogTransparent = false;
+    // used to render surface normals to G buffer for ssao effect
+    this.normalsToGBuffer = false;
+    // used for toon material
+    this.toonShading = false;
+    // set default values
+    this.setValues.call(this, {
```
Comment (AnaMoskvina) [inline packages/lib/src/gfx/shaders/UberMaterial.js:174]: can we not use context binding here?

File: packages/lib/src/gfx/shaders/UberMaterial.js [hunk 3]
```diff
@@ -197,187 +121,257 @@ UberMaterial.prototype.uberOptions = {
...
-  THREE.RawShaderMaterial.prototype.setValues.call(this, values);
+  setValues(values) {
+    if (typeof values === 'undefined') {
+      return;
+    }
 
-  const defines = {};
-  const extensions = {};
+    // set direct values
+    THREE.RawShaderMaterial.prototype.setValues.call(this, values);
```
Comment (AnaMoskvina) [inline packages/lib/src/gfx/shaders/UberMaterial.js:247]: Can we use super.setValues(values) here?

File: packages/lib/src/gfx/shaders/UberMaterial.js [hunk 4]
```diff
@@ -197,187 +121,257 @@ UberMaterial.prototype.uberOptions = {
...
+    this.setValues.call(this, {
+      uniforms: THREE.UniformsUtils.clone(defaultUniforms),
+      vertexShader: this.precisionString() + vertexShader,
+      fragmentShader: this.precisionString() + fragmentShader,
+      lights: true,
+      fog: true,
+      side: THREE.DoubleSide,
+    });
+    this.uberOptions = uberOptions;
+    this.setValues(params);
```
Comment (AnaMoskvina) [inline packages/lib/src/gfx/shaders/UberMaterial.js:183]: can we set all values with a single setValues call?

Comment (r1z3rISGOD) [reply packages/lib/src/gfx/shaders/UberMaterial.js:183]: Probably yes, we can

File: packages/lib/src/gfx/shaders/UberMaterial.js (hunk 3 above)
Comment (r1z3rISGOD) [reply packages/lib/src/gfx/shaders/UberMaterial.js:247]: Yes we can

File: packages/lib/src/gfx/shaders/UberMaterial.js (hunk 2 above)
Comment (r1z3rISGOD) [reply packages/lib/src/gfx/shaders/UberMaterial.js:174]: yes, we can

File: packages/lib/src/gfx/shaders/UberMaterial.js (hunk 1 above)
Comment (r1z3rISGOD) [reply packages/lib/src/gfx/shaders/UberMaterial.js:195]: yes, we can

**Review by AnaMoskvina — APPROVED** [review]

**Pull request merged**